gluten-data/src/main/java/io/glutenproject/backendsapi/glutendata/GlutenDataIteratorApi.scala

```diff
@@ -0,0 +1,365 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.backendsapi.glutendata
+
+import java.util
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ListBuffer
+
+import io.glutenproject.GlutenNumaBindingInfo
+import io.glutenproject.backendsapi.IIteratorApi
+import io.glutenproject.columnarbatch.ArrowColumnarBatches
+import io.glutenproject.execution._
+import io.glutenproject.memory.{GlutenMemoryConsumer, TaskMemoryMetrics}
+import io.glutenproject.memory.alloc._
+import io.glutenproject.memory.arrowalloc.ArrowBufferAllocators
+import io.glutenproject.substrait.plan.PlanNode
+import io.glutenproject.substrait.rel.LocalFilesBuilder
+import io.glutenproject.substrait.rel.LocalFilesNode.ReadFileFormat
+import io.glutenproject.utils.GlutenDataImplicitClass.{coalesce, ArrowColumnarBatchRetainer}
+import io.glutenproject.vectorized._
+import org.apache.arrow.vector.types.pojo.Schema
+
+import org.apache.spark.{InterruptibleIterator, SparkConf, SparkContext, TaskContext}
+import org.apache.spark.internal.Logging
+import org.apache.spark.memory.TaskMemoryManager
+import org.apache.spark.rdd.RDD
+import org.apache.spark.softaffinity.SoftAffinityUtil
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.connector.read.InputPartition
+import org.apache.spark.sql.execution.datasources.FilePartition
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.utils.SparkArrowUtil
+import org.apache.spark.sql.utils.OASPackageBridge.InputMetricsWrapper
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+import org.apache.spark.util.ExecutorManager
+import org.apache.spark.util.memory.TaskMemoryResources
+
+abstract class GlutenDataIteratorApi extends IIteratorApi with Logging {
```

Comment:
GlutenDataIteratorApi is a bit long, how about GlutenIteratorApi?

Reply:
I am OK to both but as some extra information, GlutenDataIteratorApi is in module `gluten-data`. `GlutenXXX` is more suitable for contents under `gluten-core` to me?

Reply:
Agreed. I'll update the PR.